src/components/Api.js

```diff
@@ -0,0 +1,117 @@
+import { isObjectEmpty } from "../utils/utils.js";
+
+export default class Api {
+  constructor({ baseUrl, authToken }) {
+    this._apiURL = baseUrl;
+    this._authToken = authToken;
+  }
+
+  apiCall({ endpoint, method, body = {} }) {
+    const opt = {
+      method,
+      headers: {
+        authorization: this._authToken,
+        "Content-Type": "application/json",
+      },
+    };
+
+    if (!isObjectEmpty(body)) opt.body = JSON.stringify(body);
+
+    return fetch(this._apiURL + endpoint, opt)
+      .then((res) => {
+        return res.ok ? res.json() : Promise.reject(`Error: ${res.statusText}`);
+      })
+      .then((json) => {
+        return json;
+      })
+      .catch((err) => {
+        console.error(err);
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`   after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.